atomsci/ddm/pipeline/perf_data.py

```diff
@@ -1674,15 +1598,15 @@ def __init__(self, model_dataset, transformers, subset, predict_probs=True, tran
 
         # DeepChem does not currently support arbitary class names in classification datasets; 
         # enforce class indices (0, 1, 2, ...) here.
-        self.class_indeces = list(set(model_dataset.dataset.y.flatten()))
+        self.real_classes = model_dataset.get_untransformed_responses(dataset.ids)
+        self.class_indeces = list(set(self.real_classes.flatten()))
         self.num_classes = len(self.class_indeces)
-        self.real_classes = dataset.y
         # Convert true values to one-hot encoding
         if self.num_classes > 2:
-            self.real_vals = np.concatenate([dc.metrics.to_one_hot(dataset.y[:,j], self.num_classes).reshape(-1,1,self.num_classes)
+            self.real_vals = np.concatenate([dc.metrics.to_one_hot(self.real_classes[:,j], self.num_classes).reshape(-1,1,self.num_classes)
                                              for j in range(self.num_tasks)], axis=1)
```

Comment:
Does one-hot encoding count as a transformation? for a refactor?

Reply:
I think sklearn lets you add a one-hot step into the pipeline.